src/py/mat3ra/made/tools/convert.py

```diff
@@ -0,0 +1,186 @@
+    """
+    __round__ = RoundNumericValuesMixin.round_array_or_number
+    basis = {
+        "elements": [{"id": i, "value": str(site.specie)} for i, site in enumerate(structure.sites)],
+        "coordinates": [
+            {"id": i, "value": __round__(list(site.frac_coords))} for i, site in enumerate(structure.sites)
+        ],
+        "units": "crystal",
+        "cell": __round__(structure.lattice.matrix.tolist()),
+        "constraints": [],
+    }
+
+    lattice = {
+        "a": __round__(structure.lattice.a),
+        "b": __round__(structure.lattice.b),
+        "c": __round__(structure.lattice.c),
+        "alpha": __round__(structure.lattice.alpha),
+        "beta": __round__(structure.lattice.beta),
+        "gamma": __round__(structure.lattice.gamma),
+        "units": {"length": "angstrom", "angle": "degree"},
+        "type": "TRI",
+        "vectors": {
+            "a": __round__(structure.lattice.matrix[0].tolist()),
+            "b": __round__(structure.lattice.matrix[1].tolist()),
+            "c": __round__(structure.lattice.matrix[2].tolist()),
+            "alat": 1,
+            "units": "angstrom",
+        },
+    }
+
+    material_data = {
+        "name": structure.formula,
+        "basis": basis,
+        "lattice": lattice,
+        "isNonPeriodic": not structure.is_ordered,
+        "_id": "",
+        "metadata": {"boundaryConditions": {"type": "pbc", "offset": 0}},
+        "isUpdated": True,
+    }
+
+    return material_data
+
+
+def to_poscar(material_or_material_data: Union[Material, Dict[str, Any]]) -> str:
+    """
+    Converts material object in ESSE format to a POSCAR string.
+
+    Args:
+        material_or_material_data (dict|class): A dictionary containing the material information in ESSE format.
+
+    Returns:
+        str: A POSCAR string.
+    """
+    structure = to_pymatgen(material_or_material_data)
+    poscar = Poscar(structure)
+    # For pymatgen `2023.6.23` supporting py3.8 the method name is "get_string"
+    # TODO: cleanup the if statement when dropping support for py3.8
+    if hasattr(poscar, "get_string"):
+        return poscar.get_string()
+    return poscar.get_str()
+
+
+def from_poscar(poscar: str) -> Dict[str, Any]:
+    """
+    Converts a POSCAR string to a material object in ESSE format.
+
+    Args:
+        poscar (str): A POSCAR string.
+
+    Returns:
+        dict: A dictionary containing the material information in ESSE format.
+    """
+    structure = Structure.from_str(poscar, "poscar")
+    return from_pymatgen(structure)
+
+
+def to_ase(material_or_material_data: Union[Material, Dict[str, Any]]) -> Atoms:
+    """
+    Converts material object in ESSE format to an ASE Atoms object.
+
+    Args:
+        material_or_material_data (dict|class): A dictionary containing the material information in ESSE format.
+
+    Returns:
+        Any: An ASE Atoms object.
+    """
+    # TODO: check that atomic labels are properly handled
```

Comment:
@VsevolodX FYSE - we will need to check that this is working b/c of the difference in Py3.8 toolchain: https://github.com/Exabyte-io/made/pull/122/files#diff-92d73dbc1fad849082fdef64b09c5b6e8d28bf7d2324a1bd220c4d3b32c2c2e1R48. If problematic, we will need to use native ASE conversion (like in to_pymatgen) instead of using the pymatgen adapter.

Reply:
Ok